tests/unit/test_library_saml.py

```diff
@@ -5,16 +5,23 @@
 from charms.saml_integrator.v0 import saml
 from ops.testing import Harness
 
-METADATA = """
+REQUIRER_METADATA = """
 name: saml-consumer
 requires:
   saml:
     interface: saml
 """
 
+PROVIDER_METADATA = """
+name: saml-producer
+provides:
+  saml:
+    interface: saml
+"""
+
 
-class SamlConsumerCharm(ops.CharmBase):
-    """Class for consumer charm testing."""
+class SamlRequirerCharm(ops.CharmBase):
```

Comment:
Could any charm be used here @weiiwang01?

Reply:
Yes, potentially. I though that given the simplicity of this scenario this was simpler

Reply:
I think this is an unit test, not an integration test. So any-charm would not be very useful here

Reply:
I though it worked for both unit and integration. My bad. Thanks for the explanation